storage/client/queue/src/samples/java/com/azure/storage/queue/AsyncSamples.java

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.queue;
+
+import java.util.UUID;
+
+/*
+ *  This example mimics some arbitrary number of clients continuously sending messages up to a queue in a parallel and
+ *  a server dequeuing the messages and processing them.
+ */
+public class AsyncSamples {
+    private static final String ACCOUNT_NAME = System.getenv("AZURE_STORAGE_ACCOUNT_NAME");
+    private static final String SAS_TOKEN = System.getenv("PRIMARY_SAS_TOKEN");
+    private static final String QUEUE_NAME = generateRandomName("async-call", 16);
+
+    /**
+     * The main method shows how we do the basic operations of enqueueing and dequeueing messages on async queue client.
+     * @param args No args needed for main method.
+     */
+    public static void main(String[] args) {
+        // Create an async queue client.
+        String queueURL = String.format("https://%s.queue.core.windows.net/%s%s", ACCOUNT_NAME, QUEUE_NAME, SAS_TOKEN);
+        QueueAsyncClient queueAsyncClient = new QueueClientBuilder().endpoint(queueURL).buildAsyncClient();
+
+        // Create a queue, enqueue two messages.
+        queueAsyncClient.create()
+            .flatMap(response -> queueAsyncClient.enqueueMessage("This is message 1"))
```

Comment:
This doesn't seem to be the correct way to chain Reactor requests, I think `then()` should be used

Reply:
Make sense. Did not turn Mono to Flux and next operation did not depend on the response from the async request. I think doOnSuccess fits the situation as we need to have a queue before enqueue the message.

Reply:
Will change the first one to doOnSuccess, and second one change to then.